### DISCUSSION TITLE
ファームウェア書き込み後認識がされない

### DISCUSSION THREAD

**Discussion by hoshiboshino** [Yowkees/keyball#287] [Q&A] [answered]
操作誤りで同じ投稿が複数されてしまい申し訳ありません。

keyball39の挙動についてご教示ください。
ビルドガイドの「3-7．ProMicroの書き込みと動作確認」でREMAPに登録してあるKeyball_test.hexの書き込みを行いましたが、キー入力が一切行えません。
テストファームウエアでも正規ファームウエアのどちらでも同じ症状が発生します。

デバイスマネージャーを確認したところ、
リセットボタンを2回素早く押した直後「ポートCOMとLPT）」にUSB Serial Device(com3）として認識され、8秒程度すると一覧から消えてしまいます。

ファームウェアのかきこみは正常に終了しているように見えるので組付けの問題なのでしょうか。
![P_20230723_185252](https://github.com/Yowkees/keyball/assets/140311907/a859031c-15b6-44f8-a951-e6571e736d8b)

以下のQAと一部同じですが、ドックステーションのしようはしておらず、promicro単体で正規ファームウェアをかきこんでから組み付けても症状が改善されません。
https://github.com/Yowkees/keyball/discussions/244#discussion-5251301
https://github.com/Yowkees/keyball/discussions/259#discussion-5327229

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

まったくキー入力ができないということで了解しました。順に確認したいと思います。
リセット時の挙動は、８秒後にリセットモードが解除されているだけですので正常と思います。

キー入力はできないということですが、OLEDは表示されていますか？トラックボールを動かしたときやキー入力したときにOLED表示は変化するでしょうか？

また、Keyball39基板の半田付け面のお写真も見せていただければ幸いです。何かアドバイスできるかもしれません。

以上、お手数ですがご確認お願いいたします。

**Comment by hoshiboshino** [comment]
ご連絡いただきありがとうございます。
OLEDは「keyball」の表示のままキーを押下してもトラックボールを操作しても表示に変化がありません。
12ピンソケット取付時に、いくつかの足にハンダを流し込みすぎてしまいピンヘッダーがきれいに刺さらず、刺さるところまで指してpromicroを取り付けたため、うまく電極が接触できていない可能性もあるかなとおもっております。

基盤面の写真を添付いたします。お手数ですがご確認のほどよろしくお願い致します。

**Comment by hoshiboshino** [comment]
![P_20230724_120816](https://github.com/Yowkees/keyball/assets/140311907/a419ba02-50c6-414b-b4e4-6516eedbcf3f)
![P_20230724_121234](https://github.com/Yowkees/keyball/assets/140311907/eac92876-a6ba-449f-8f4d-3287426a1172)

**Comment by Yowkees** [comment]
お写真ありがとうございます。半田付け忘れや半田付け位置のミス等はなさそうですね。

OLEDは両手ともKeyballロゴが表示されているということでしょうか？
その場合、うまくPC側からUSBデバイスとして認識されていない状態です。
可能でしたら、USBケーブルやPCを変えてどのような挙動になるかご確認いただけませんでしょうか。

お手数ですがよろしくお願いいたします。

**Comment by hoshiboshino** [comment]
追加のご確認ありがとうございます。
異なるPC(以下 B端末 と記載）に接続してみたところkeyball39として認識し、トラックボール及びキーボードともに動作いたしました。
![P_20230724_181454](https://github.com/Yowkees/keyball/assets/140311907/930be4fa-655c-451b-af73-80db92e3e8cd)

いくつかのキーが認識しないため、ソケットとダイオードの確認をおこなってみます。

B端末で動作確認後、keyballとして認識してくれないPC(以下 A端末 と記載）に接続を行いましたが依然としてUSBデバイスとして認識がされません。
A端末においてどのような対応を行えばUSBデバイスとして認識されるようになりますでしょうか。

**Comment by Yowkees** [comment]
通常は必要ないのですが、Arduinoの最新のドライバーをインストールすると改善できるかもしれません。
また、12Pピンソケット内に半田が流れ込みピンヘッダーが差さりきらない状態ですと、今はかろうじて導通している状態ですので、やはりピンソケットを新品に交換したうえで検証することをお勧めします。

以上、お手数ですがご検討よろしくお願いいたします。

**Comment by hoshiboshino** [comment]
ご回答いただきありがとうございます。
A端末においてUSBデバイスとして認識されないのはSS対応のUSBポートを使用していたためでした。SSの記載のないポートを使用したところ、問題なく動作致しました。

またピンソケットについてもコメントいただきありがとうございます。現在ピンソケット及びピンヘッダーを注文しており、届き次第換装して改めて動作確認を行ってみます。

ご支援賜り誠にありがとうございました。

**Comment by Yowkees** [comment, marked as answer]
いろいろとご調査いただきありがとうございました。USBポートでしたか。
ぜひピンソケットを交換いただき、ご愛用いただければ幸いです。
今後ともよろしくお願いいたします。
